docs/modules/demos/pages/jupyterhub-keycloak.adoc

```diff
@@ -0,0 +1,197 @@
+In the example below that would then be 172.19.0.5:31095:
+
+[source,yaml]
+----
+apiVersion: v1
+data:
+  keycloakAddress: 172.19.0.5:31093 # Keycloak itself
+  keycloakNodeIp: 172.19.0.5 # can be used to access the proxy-public service
+kind: ConfigMap
+metadata:
+  name: keycloak-address
+  namespace: default
+----
+
+NOTE: The hub may show `CreateContainerConfigError`` for a few moments on start-up as it requires the ConfigMap written by the Keycloak deployment.
+
+You should see the JupyterHub login page, which will indicate a re-direct to the OAuth service (Keycloak):
+
+image::jupyterhub-keycloak/oauth-login.png[]
+
+Click on the sign-in button.
+You will redirected to the Keycloak login, where you can enter one of the afore-mentioned users:
+
+image::jupyterhub-keycloak/keycloak-login.png[]
+
+A successful login will redirect you back to JupyterHub where different profiles are listed (the drop-down options are visible when you click on the respective fields):
+
+image::jupyterhub-keycloak/server-options.png[]
+
+The explorer window on the left includes HDFS configuration files and a notebook that is already mounted.
+Open the `hdfs` folder where the `core-site.xml` and `hdfs-site.xml` from the discovery ConfigMap of the HDFS cluster are located.
+You can see how to reference the HDFS cluster when reading from it in the notebook (`hdfs://hdfs/`):
+
+image::jupyterhub-keycloak/hdfs-core-site.png[]
+
+Double-click on the file `notebook/process-s3.ipynb`:
+
+image::jupyterhub-keycloak/load-nb.png[]
+
+Run the notebook by selecting "Run All Cells" from the menu:
+
+image::jupyterhub-keycloak/run-nb.png[]
+
+The notebook includes some comments regarding image compatibility and uses a custom image built off the official Spark official that matches the Spark version used in the notebook.
+The java versions also match exactly.
+Python versions need to match at the `major:minor` level, which is why Python 3.11 is used in the custom image.
+
+Once the spark executor has been started (we have specified `spark.executor.instances` = 1) it will spin up as an extra pod.
+We have named the spark job to incorporate the current user (justin-martin).
+JupyterHub has started a pod for the user's notebook instance (`jupyter-justin-martin---bdd3b4a1`) and another one for the spark executor (`process-s3-jupyter-justin-martin-bdd3b4a1-9e9da995473f481f-exec-1`):
+
+[source,console]
+----
+12:49 $ kubectl get pods
+NAME                                   READY   STATUS      RESTARTS   AGE
+...
+jupyter-justin-martin---bdd3b4a1       1/1     Running     0          17m
+process-s3-jupyter-justin-martin-...   1/1     Running     0          2m9s
+...
+----
+
+Stop the kernel in the notebook (which will shut down the spark session and thus the executor) and log out as the current user.
+Log in now as `daniel.king` and then again `isla.williams` (you may need so do this in a clean browser sessions so that existing login cookies are removed).
```

Comment:
```suggestion
Log in now as `daniel.king` and then again as `isla.williams` (you may need to do this in a clean browser sessions so that existing login cookies are removed).
```